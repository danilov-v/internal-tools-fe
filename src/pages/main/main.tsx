import React, { useEffect } from 'react';
import { RouteComponentProps, Router, useNavigate } from '@reach/router';
import { useSelector, useDispatch } from 'react-redux';
import { getProfile } from 'redux/profile/thunks';
import {
  isAuthChecked as isAuthCheckedSelector,
  getProfileInfo,
} from 'redux/profile/selectors';
import { Header } from 'components/Header';
import { SoldersList } from 'components/SoldersList';
import { PersonnelDetails } from 'pages/personnelDetails/personnelDetails';

import { LoadingScreen } from 'pages/main/components/LoadingScreen';
import { NotFound } from 'pages/main/components/NotFound';

const Main: React.FC<RouteComponentProps> = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const profileInfo = useSelector(getProfileInfo);
  const isAuthChecked = useSelector(isAuthCheckedSelector);

  useEffect(() => {
    if (!isAuthChecked) {
      dispatch(getProfile());
    }
  }, [dispatch, isAuthChecked]);

  if (!isAuthChecked) {
    return <LoadingScreen />;
  }

  if (isAuthChecked && !profileInfo) {
    navigate('sign-in');
    return null;
  }

  return (
    <>
      <Header />
      <Router>
        <NotFound default />
        <SoldersList path="personnel" />
        <PersonnelDetails path="personnel-details/:id" />
      </Router>
    </>
  );
};

export { Main };
