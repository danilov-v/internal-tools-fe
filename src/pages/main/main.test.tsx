import React from 'react';
import { render } from '@testing-library/react';

import { Main } from './main';

describe('pages/main', () => {
  it('should render and match snapshot', () => {
    const page = render(<Main />);

    expect(page.baseElement).toMatchSnapshot();
  });
});
