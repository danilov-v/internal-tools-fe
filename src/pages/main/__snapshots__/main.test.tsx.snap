// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/main should render and match snapshot 1`] = `
<body>
  <div>
    <header
      class="sc-AxgMl gxzYZF"
    >
      <div
        class="sc-AxjAm cbGFkU"
      >
        <img
          alt="logo"
          class="sc-AxirZ dbNCkZ"
          src="logo.svg"
        />
      </div>
      <ul
        class="sc-AxheI fGHfXX"
      >
        <li
          class="sc-Axmtr ijesAw"
        >
          <a
            class="sc-AxmLO byZQTA"
            href="/main"
            state="not-active"
          >
            Рота
          </a>
        </li>
        <li
          class="sc-Axmtr ijesAw"
        >
          <a
            class="sc-AxmLO byZQTA"
            href="/#schedule"
            state="not-active"
          >
            График
          </a>
        </li>
      </ul>
      <div
        class="sc-fzoLsD ejQvyh"
      >
        <div
          class="sc-AxiKw bsHBHU"
        >
          <img
            alt="Avatar"
            class="sc-AxhCb fVlVaN"
            src="captain_rang.svg"
          />
          <div
            class="sc-AxhUy dOLKgI"
          >
            <span>
              Командир роты
            </span>
            Стороженко
          </div>
        </div>
      </div>
      <div
        class="sc-fzozJi cbhuUS"
      >
        <div />
        <div />
        <div />
      </div>
    </header>
  </div>
</body>
`;
